ext/pcre/php_pcre.c

```diff
@@ -2072,58 +2072,55 @@ static zend_always_inline zend_string *php_pcre_replace_func(zend_string *regex,
 
 /* {{{ php_pcre_replace_array
  */
-static zend_string *php_pcre_replace_array(HashTable *regex, zval *replace, zend_string *subject_str, size_t limit, size_t *replace_count)
+static zend_string *php_pcre_replace_array(HashTable *regex,
+	zend_string *replace_str, HashTable *replace_ht,
+	zend_string *subject, size_t limit, size_t *replace_count)
 {
 	zval		*regex_entry;
 	zend_string *result;
-	zend_string *replace_str, *tmp_replace_str;
+	zend_string *subject_str = zend_string_copy(subject);
```

Comment:
You can use one variable and zend_string_addref (zend_string_copy = zend_string_addref + return).